lib/pure/streams.nim

```diff
@@ -176,11 +176,11 @@ proc peekChar*(s: Stream): char =
 
 proc readBool*(s: Stream): bool =
   ## reads a bool from the stream `s`. Raises `IOError` if an error occurred.
-  read(s, result)
+  readData(s, addr(result), sizeof(bool)) != 0
```

Comment:
Repasting my comment from https://github.com/narimiran/Nim/commit/f1c5e7aa52c732cefec0183149cdcc85514e9cc7#r33227653

--------

I think the fix is wrong.

readData will write the bool read with `addr(result)` and returns and error code.
And then this error code is compared to 0 and that result is written as the read bool, this should be where exception is thrown instead.

The issue is that `a` binary representation is 01100001 instead of 00000001 (true) or 00000000 (false) and C doesn't have a real notion of bool.

Instead we should do something like:

```Nim
proc readBool*(s: Stream): bool =
  ## reads a bool from the stream `s`. Raises `IOError` if an error occurred.
  var tmp: byte
  read(s, tmp)
  result = tmp != byte 0
```

Also I think it's important to say that we read a stream byte by byte in case of bool as some might assume that for bool we read bit by bit.

Reply:
To avoid that I think there should be a test of reading a 0 string added as well.